tests/unit/test_characterProcessing.py

```diff
@@ -0,0 +1,308 @@
+		triple="111,111"),
+	NrProcTestCase(origin="1111,1111",
+		single="1  1  1  1,1  1  1  1",
+		double="11  11,11  11",
+		triple="1  111,1  111"),
+	# Decimal with letters
+	NrProcTestCase(origin="1a,234b",
+		single="1a,2  3  4b",
+		double="1a,2  34b",
+		triple="1a,234b"),
+	NrProcTestCase(origin="1a,2,345b",
+		single="1a,2,3  4  5b",
+		double="1a,2,3  45b",
+		triple="1a,2,345b"),
+	NrProcTestCase(origin="1a,2,3456b",
+		single="1a,2,3  4  5  6b",
+		double="1a,2,34  56b",
+		triple="1a,2,3  456b"),
+
+]
+
+thousandsSepDot = [
+	NrProcTestCase(origin="1.234",
+		single="1  2  3  4",
+		double="1  2  34",
+		triple="1  234"),
+	NrProcTestCase(origin="12.345",
+		single="1  2  3  4  5",
+		double="12  3  45",
+		triple="12  345"),
+	NrProcTestCase(origin="123.456",
+		single="1  2  3  4  5  6",
+		double="1  23  4  56",
+		triple="123  456"),
+	NrProcTestCase(origin="1.234.567",
+		single="1  2  3  4  5  6  7",
+		double="1  2  34  5  67",
+		triple="1  234  567"),
+	# Thousands separator with letters
+	NrProcTestCase(origin="1a.234b",
+		single="1a.2  3  4b",
+		double="1a.2  34b",
+		triple="1a.234b"),
+	NrProcTestCase(origin="1a.2.345b",
+		single="1a.2  3  4  5b",
+		double="1a.2  3  45b",
+		triple="1a.2  345b"),
+]
+
+thousandsSepComma = [
+	NrProcTestCase(origin="1,234",
+		single="1  2  3  4",
+		double="1  2  34",
+		triple="1  234"),
+	NrProcTestCase(origin="12,345",
+		single="1  2  3  4  5",
+		double="12  3  45",
+		triple="12  345"),
+	NrProcTestCase(origin="123,456",
+		single="1  2  3  4  5  6",
+		double="1  23  4  56",
+		triple="123  456"),
+	NrProcTestCase(origin="1,234,567",
+		single="1  2  3  4  5  6  7",
+		double="1  2  34  5  67",
+		triple="1  234  567"),
+	# Thousands separator with letters
+	NrProcTestCase(origin="1a,234b",
+		single="1a,2  3  4b",
+		double="1a,2  34b",
+		triple="1a,234b"),
+	NrProcTestCase(origin="1a,2,345b",
+		single="1a,2  3  4  5b",
+		double="1a,2  3  45b",
+		triple="1a,2  345b"),
+]
+
+thousandsSepNBSP = [
+	NrProcTestCase(origin="1\xa0234",
+		single="1  2  3  4",
+		double="1  2  34",
+		triple="1  234"),
+	NrProcTestCase(origin="12\xa0345",
+		single="1  2  3  4  5",
+		double="12  3  45",
+		triple="12  345"),
+	NrProcTestCase(origin="123\xa0456",
+		single="1  2  3  4  5  6",
+		double="1  23  4  56",
+		triple="123  456"),
+	NrProcTestCase(origin="1\xa0234\xa0567",
+		single="1  2  3  4  5  6  7",
+		double="1  2  34  5  67",
+		triple="1  234  567"),
+	# Thousands separator with letters
+	NrProcTestCase(origin="1a\xa0234b",
+		single="1a\xa02  3  4b",
+		double="1a\xa02  34b",
+		triple="1a\xa0234b"),
+	NrProcTestCase(origin="1a\xa02\xa0345b",
+		single="1a\xa02  3  4  5b",
+		double="1a\xa02  3  45b",
+		triple="1a\xa02  345b"),
+]
+
+# English
+nrProcTestCases["en"] = onlyDigits + decimalDot + thousandsSepComma
+
+# Dutch
+nrProcTestCases["nl"] = onlyDigits + decimalComma + thousandsSepDot
+
+# Swedish
+nrProcTestCases["sv"] = onlyDigits + decimalComma + thousandsSepNBSP
```

Comment:
Is it the case that some of these languages use multiple separators depending on context?